src/Stratis.SmartContracts.Core.Validation/Validators/Module/SingleTypeValidator.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using Stratis.ModuleValidation.Net;
+
+namespace Stratis.SmartContracts.Core.Validation
+{
+    /// <summary>
+    /// Validates that a <see cref="Mono.Cecil.ModuleDefinition"/> contains a single Type
+    /// </summary>
+    public class SingleTypeValidator : IModuleDefinitionValidator
+    {
+        private static readonly HashSet<string> IgnoredInCount = new HashSet<string>
+        {
+            "<Module>", // Part of every module
+            "<PrivateImplementationDetails>" // Added when constructing an array
+        };
+
+        public IEnumerable<ValidationResult> Validate(ModuleDefinition module)
+        {
+            List<TypeDefinition> typeDefinitions = module.Types.Where(x => !IgnoredInCount.Contains(x.FullName)).ToList();
```

Comment:
The `if` below can be combined in the `where`? Also pretty sure we don't need to the `ToList()`